Lib/unittest/test/test_case.py

```diff
@@ -1950,5 +1953,234 @@ def test2(self):
+        expect = ('- foo', '+ bar')
+        self.assertHeuristicDiffReturns(a, b, expect)
+
+    def test_unified_diff_is_used_with_large_inputs(self):
+        """One long line, as well as many single-character lines."""
+        # one long line
+        a = ('foo' * self.N,)
+        b = ('bar' * self.N,)
+        expect = ('--- expected\n', '+++ got\n', '@@ -1 +1 @@\n',
+                  '-' + 'foo' * self.N,
+                  '+' + 'bar' * self.N)
+        self.assertHeuristicDiffReturns(a, b, expect)
+
+        # many lines
+        a = ('1\n' * self.N).splitlines()
+        b = ('2\n' * self.N).splitlines()
+        expect = ('--- expected\n', '+++ got\n', f'@@ -1,{self.N} +1,{self.N} @@\n',
+                  *(['-1'] * self.N),
+                  *(['+2'] * self.N))
+        self.assertHeuristicDiffReturns(a, b, expect)
+
+    def test_ndiff_to_unified_diff_breaking_point_long_line(self):
+        """This is the approximate single line length at which the heuristic
+        will switch from ndiff to unified_diff."""
+        expect_switch_at = 125_000
+        a = ''
+        b = ''
+        n = expect_switch_at // 2
+        while '@' not in ''.join(unittest.case._heuristic_diff(a, b)):  # type: ignore
+            n *= 1.3
+            a = ('a' * int(n),)
+            b = ('b' * int(n),)
+
+        self.assertGreater(n, expect_switch_at * 0.8)
+        self.assertLess(n, expect_switch_at * 1.1)
+
+    def test_ndiff_to_unified_diff_breaking_point_many_lines(self):
+        """For lines just one character long, the heuristic will switch from
+        ndiff to unified_diff around 70,000 differing lines."""
+        expect_switch_at = 70_000
+        a = ''
+        b = ''
+        n = expect_switch_at // 2
+        while '@' not in ''.join(unittest.case._heuristic_diff(a, b)):  # type: ignore
+            n *= 1.3
+            a = ('a\n' * int(n),)
+            b = ('b\n' * int(n),)
+
+        self.assertGreater(n, expect_switch_at * 0.8)
+        self.assertLess(n, expect_switch_at * 1.1)
+
+    def test_ndiff_to_unified_diff_scaled_line_and_cols(self):
+        """Scale line length and number of differing columns at different
+        rates, expecting a switch to `unified_diff` at specified points.
+        """
+
+        @dataclass
+        class Case:
+            """Case class specifies parameters for all tests."""
+            line_length_factor: int
+            num_lines_factor: int
+            extent_differing: float
+
+            # this is a "magic number" for all cases where the heuristic
+            # will switch from using ndiff to unified_diff.
+            expect_unified_diff_at: int
+
+        # --- Layout test cases
+        # ---------------------
+
+        cases = (
+            # scale width and length by ratios of 2:1
+            Case(
+                line_length_factor=1,
+                num_lines_factor=2,
+                extent_differing=1,
+                expect_unified_diff_at = 22,
```

Comment:
```suggestion
                expect_unified_diff_at=22,
```
Consistent formatting (also reappears several times)

Reply:
This one wasn't changed yet

Reply:
@jdevries3133 does this give the right context? This conversation was hidden by default so maybe you missed it.